python/python/bystro/ancestry/inference.py

```diff
@@ -161,25 +178,60 @@ def infer_ancestry(ancestry_model: AncestryModel, genotypes: Dataset) -> Ancestr
 
     logger.debug("Beginning ancestry inference")
 
+    logger.info(
+        "Memory usage before dosage matrix filtered row counting: %s (MB)",
+        psutil.Process(os.getpid()).memory_info().rss / 1024**2,
+    )
+
     with Timer() as timer:
-        mask = pc.field("locus").isin(ancestry_model.pca_loadings_df.index)
-        scanner = genotypes.scanner(filter=mask)
+        gnomad_model = ancestry_models.gnomad_model
+        array_model = ancestry_models.array_model
 
-        # Initialize an empty DataFrame or a list to collect batches
-        filtered_genotypes = []
+        mask_gnomad = pc.field("locus").isin(gnomad_model.pca_loadings_df.index)
+        mask_array = pc.field("locus").isin(array_model.pca_loadings_df.index)
 
-        for batch in scanner.to_batches():
-            # Convert the current batch to a pandas DataFrame
-            batch_df = batch.to_pandas()
+        scanner_gnomad = genotypes.scanner(filter=mask_gnomad)
+        scanner_array = genotypes.scanner(filter=mask_array)
 
-            # Set the index to 'locus', assuming 'locus' is a column in your dataset
-            batch_df = batch_df.set_index("locus")
+        gnomad_matching_row_count = scanner_gnomad.count_rows()
+        array_matching_row_count = scanner_array.count_rows()
 
-            # Append the processed batch to the list
-            filtered_genotypes.append(batch_df)
+        logger.debug(
+            "Found %d rows in genotypes matching gnomAD PCA loadings",
+            gnomad_matching_row_count,
+        )
+        logger.debug(
+            "Found %d rows in genotypes matching array PCA loadings",
+            array_matching_row_count,
+        )
+
+        logger.info(
+            "Memory usage after dosage matrix filtered row counting: %s (MB)",
+            psutil.Process(os.getpid()).memory_info().rss / 1024**2,
+        )
+
+        if gnomad_matching_row_count >= array_matching_row_count:
```

Comment:
Since the array set has about half of the gnomad set (~34k vs ~74k), the gnomad set could beat the array set even in a case where the percentage is 100% variant match to the array set with just 50% variant match to the gnomad set. Could we adjust it to take the highest percent retention instead of highest row count?

Reply:
@austinTalbot7241993 what do you think? Would it be better to use absolute number of snps / informative PC's, or completeness respective to the model (or a hybrid strategy)?

Reply:
Based on Dave's comments, if you're open to it, unless Austin strongly feels otherwise, let's leave as is, and then run the test? That way we don't block deployment of something that is a strict improvement over what we had before, and can optimize in a subsequent PR?